corehq/motech/generic_inbound/models.py

```diff
@@ -72,3 +83,61 @@ def to_json(self):
             "expression_id": self.expression_id,
             "message": self.message,
         }
+
+
+class RequestLog(models.Model):
+
+    class Status(models.TextChoices):
+        FILTERED = 'filtered', _('Filtered')
+        VALIDATION_FAILED = 'validation_failed', _('Validation Failed')
+        SUCCESS = 'success', _('Success')
+        ERROR = 'error', _('Error')
+        REVERTED = 'reverted', _('Reverted')
+
+    class RequestMethod(models.TextChoices):
+        POST = 'POST'
+        PUT = 'PUT'
+        PATCH = 'PATCH'
+
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    domain = models.CharField(max_length=255)
+    api = models.ForeignKey(ConfigurableAPI, on_delete=models.CASCADE)
+    status = models.CharField(max_length=32, choices=Status.choices)
+
+    timestamp = models.DateTimeField(auto_now=True, db_index=True)
+    attempts = models.PositiveSmallIntegerField(default=1)
+    response_status = models.PositiveSmallIntegerField()
+    error_message = models.TextField()
+
+    username = models.CharField(max_length=128)
+    request_method = models.CharField(max_length=32, choices=RequestMethod.choices)
+    request_query = models.CharField(max_length=8192)
+    request_body = models.TextField()
+    request_headers = models.JSONField(default=dict)
+    request_user_agent = models.TextField()
```

Comment:
why is this separate from the headers?

Reply:
I made it a top level field since it'll be needed for reprocessing, but I suppose that can also be extracted from headers at that point, and it probably won't be filtered on.  I'll remove it.

Reply:
Oh actually it looks like it's not one of the auditcare standard request headers - auditcare normalizes user agent:
https://github.com/dimagi/commcare-hq/blob/d691d5b87d09aea74ef99c67065c606d3c0e8198/corehq/apps/auditcare/models.py#L73-L75
Sounds like options are:

* Keep it separate, as done here
* Follow auditcare's lead and split it out
* Add user agent to the list of "standard" headers and specifically exclude it from the auditcare use-case.

None of those sounds like an obviously awesome solution, but I think I favor #3, just pushed a go at that, but feel free to second-guess.

Reply:
Option 3 seems fine to me. Thanks